### PR TITLE
HW8_Vladyslav_Gurevych

### PR DESCRIPTION
**Pull request by VladyslavGurevych**

### CODE REVIEW

File: Vladyslav_Gurevych/HW8/All.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+def own_all(data: list):
```
Comment (bohdandovbsyh) [inline Vladyslav_Gurevych/HW8/All.py]: image
Try to avoid such naming

Comment (bohdandovbsyh) [inline Vladyslav_Gurevych/HW8/All.py]: add annotation for return -> bool

File: Vladyslav_Gurevych/HW8/Filter.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        return True
+
+
+if __name__ == "__main__":
+    example = [3.123, "1", 2, [], (), 5.7, "sksdskxksxn", 5643]
+
+    print(own_filter(is_it_digit, example))
+    print(own_filter(is_it_string, example))
+    print(own_filter(is_it_float, example))
+    print(own_filter(is_it_true, example))
```
Comment (bohdandovbsyh) [inline Vladyslav_Gurevych/HW8/Filter.py:57]: good. But actually, you can use lambda in the future except writing all these functions is_it_digit, is_it_string ....
But it is okay. Good job

File: Vladyslav_Gurevych/HW8/MINorMAX.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+def own_max(data: list, amount_of_result=1):
+    """
+       This function takes arguments(list and amount_of_result) and returns a list with a max value.
+       The number of output values can be changed by entering amount_of_result.
+    """
+    result = []
+    temp = data[:]
+    for count in range(amount_of_result):
```
Comment (bohdandovbsyh) [inline Vladyslav_Gurevych/HW8/MINorMAX.py]: count never used. When we have such variable in loop , we should use _ to show this

for _ in range(amount_of_result):

File: Vladyslav_Gurevych/HW8/MINorMAX.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+
+def own_min(data: list, amount_of_result=1):
+    """
+        This function takes arguments(list and amount_of_result) and returns a list with a max value.
+        The number of output values can be changed by entering amount_of_result.
+    """
+    result = []
+    temp = data[:]
+    for count in range(amount_of_result):
```
Comment (bohdandovbsyh) [inline Vladyslav_Gurevych/HW8/MINorMAX.py]: same here

File: Vladyslav_Gurevych/HW8/MINorMAX.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            if max_digit < temp[i]:
+                max_digit = temp[i]
+            else:
+                continue
+        result.append(max_digit)
+        temp.remove(max_digit)
+    return result
+
+
+def own_min(data: list, amount_of_result=1):
```
Comment (bohdandovbsyh) [inline Vladyslav_Gurevych/HW8/MINorMAX.py]: annotations for amount_of_result.
amount_of_result: int = 1

and for return statment

File: Vladyslav_Gurevych/HW8/MINorMAX.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
+def own_max(data: list, amount_of_result=1):
```
Comment (bohdandovbsyh) [inline Vladyslav_Gurevych/HW8/MINorMAX.py]: annotations for amount_of_result.
amount_of_result: int = 1

and for return statment

File: Vladyslav_Gurevych/HW8/Map.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
+def own_map(callback, data: list):
```
Comment (bohdandovbsyh) [inline Vladyslav_Gurevych/HW8/Map.py]: annotation for a callback. 

from typing import Callable

callback: Callable

File: Vladyslav_Gurevych/HW8/All.py (hunk 1 above)
Comment (VladyslavGurevych) [reply Vladyslav_Gurevych/HW8/All.py]: fixed in all tasks

Comment (VladyslavGurevych) [reply Vladyslav_Gurevych/HW8/All.py]: added in all tasks

File: Vladyslav_Gurevych/HW8/Filter.py (hunk 2 above)
Comment (VladyslavGurevych) [reply Vladyslav_Gurevych/HW8/Filter.py:57]: +

Comment (VladyslavGurevych) [reply Vladyslav_Gurevych/HW8/Filter.py:57]: +

File: Vladyslav_Gurevych/HW8/MINorMAX.py (hunk 3 above)
Comment (VladyslavGurevych) [reply Vladyslav_Gurevych/HW8/MINorMAX.py]: fixed

File: Vladyslav_Gurevych/HW8/MINorMAX.py (hunk 4 above)
Comment (VladyslavGurevych) [reply Vladyslav_Gurevych/HW8/MINorMAX.py]: +

File: Vladyslav_Gurevych/HW8/MINorMAX.py (hunk 5 above)
Comment (VladyslavGurevych) [reply Vladyslav_Gurevych/HW8/MINorMAX.py]: +

File: Vladyslav_Gurevych/HW8/Map.py (hunk 7 above)
Comment (VladyslavGurevych) [reply Vladyslav_Gurevych/HW8/Map.py]: added in all tasks

File: Vladyslav_Gurevych/HW8/MINorMAX.py (hunk 6 above)
Comment (VladyslavGurevych) [reply Vladyslav_Gurevych/HW8/MINorMAX.py]: +

**Review by bohdandovbsyh — APPROVED** [review]

**Pull request merged**